test/test_binary_ufuncs.py

```diff
@@ -3519,6 +3519,16 @@ def test_lerp_lowp_cpu(self, device, dtype):
                 expected = torch.lerp(xref, yref, wref).to(dtype)
                 self.assertEqual(actual, expected, atol=0.0, rtol=0.0)
 
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    def test_lerp_tensor_type_promotion(self, device, dtype):
+        start = make_tensor((5, 5), dtype=dtype, device=device, low=1, high=100)
+        end = make_tensor((5, 5), dtype=dtype, device=device, low=1, high=100)
+        weight = make_tensor((5, 5), dtype=torch.float, device=device, low=1, high=100)
```

Comment:
The original issue is only regarding the scalar tensor, so we want the test case to reflect this.
```suggestion
        weight = torch.rand((), dtype=torch.float, device=device)
```

As such, we also want to apply the minimal amount of code to enable the precise use case to pass. For example, we don't want to delete the TORCH_CHECK entirely, we only want to disable it and type promote if weight is 0 dimensional. This is because we don't actually want to affect the type promotion of lerp if the weight is a full Tensor (not a scalar). 

There were recent changes to enable lerp to have CPU scalar tensors with CUDA. I would expect this change to reuse that path as much as possible.